### ISSUE TITLE
Trigger live validation in async fashion

### ISSUE THREAD

**Issue by tortmayr** [eclipse-glsp/glsp#1408]
Currently live validation is executed as part of the `ModelSubmissionHandler.submitModelDirectly` method.
If a validator is present, we invoke validation and send the resulting `SetMarkersAction` in combination with the `SetModelAction`/`UpdateModelAction`.
This sync approach means that we potentially significantly delay the entire model update process if the live validation is expensive.

A better approach would be to send a `RequestMarkersAction` instead of directly triggering the validation. This would mean that the model update is processed directly and validation is triggered in async fashion.

In addition, we probably should send a progress action to indicate to the user that validation is still in progress.

**Comment by ivy-lli** [comment]
Thanks for creating an issue for this. 

In our codebase I solved it by send the validations in a CompletableFuture to the client, something like:
```java
CompletableFuture.runAsync(() -> {
  dispatcher.dispatch(new SetMarkersAction(validate(), MarkersReason.LIVE));
});
```

Sadly the ModelSubmissionHandler is not easy customizable (or I didn't get how to do it), so I override the RequestModelActionHandler and the OperationActionHandler to trigger this async validation.

I'm not sure about the progress messages though. I think for the initial load this could make sense, but validations are also triggered on each operation and I think it wouldn't make much sense to show a progress message on each operation.

**Comment by tortmayr** [comment]
> Sadly the ModelSubmissionHandler is not easy customizable (or I didn't get how to do it), so I override the RequestModelActionHandler and the OperationActionHandler to trigger this async validation.

Yeah, seems like the ModelSubmissionHandler is configured via annotation instead of an explicit bind-method in the DiagramModule. Nevertheless you can simply customize it like this:

```java
   @Override
   protected void configureAdditionals() {
      bind(ModelSubmissionHandler.class).to(MyModelSubmissionHandler.class).in(Singleton.class);
   }
 ```

**Comment by martin-fleck-at** [comment]
> I'm not sure about the progress messages though. I think for the initial load this could make sense, but validations are also triggered on each operation and I think it wouldn't make much sense to show a progress message on each operation.

I'm not entirely sure about that. Configuring Live Validation I would assume that the user is properly notified about the current validation state of the model as they do changes. If the live validation takes a very long time, we'd suggest to the user that the model is valid until the validation is complete which is simply untrue. That is why I'm more in favor of showing that the validation is in progress. What do you think?
